connectivity/service/src/main/java/org/eclipse/ditto/connectivity/service/mapping/CloudEventsMapper.java

```diff
@@ -0,0 +1,166 @@
+  final String TYPE = "type";
+
+  public String base64decoding(final String base64Message) throws UnsupportedEncodingException {
+    byte[] messageByte = Base64.getDecoder().decode(base64Message);
+    String decodedString = new String(messageByte);
+    return decodedString;
+  }
+
+  boolean validatePayload(String payload) {
+    Map<String, Object> incomingMessagePayload = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+    final JsonObject jsonObject = JsonFactory.newObject(payload);
+    List<JsonKey> key = jsonObject.getKeys();
+    for (JsonKey keyIterator : key) {
+      Object value = jsonObject.getValue(keyIterator).orElse(null);
+      if (value != null) {
+        incomingMessagePayload.put(keyIterator.toString(), value);
+      }
+    }
+    if (incomingMessagePayload.get("specversion") == null
+        || incomingMessagePayload.get("id") == null
+        || incomingMessagePayload.get("source") == null
+        || incomingMessagePayload.get("type") == null) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  private JsonifiableAdaptable extractData(final String message)
+      throws UnsupportedEncodingException {
+    final JsonObject payloadJson = JsonFactory.newObject(message);
+    if (payloadJson.getValue("data_base64").isPresent()) {
+      String base64Data = payloadJson.getValue("data_base64").orElse(null).asString();
+      String decodedData = base64decoding(base64Data);
+
+      final JsonifiableAdaptable decodedJsonifiableAdaptable =
+          DittoJsonException.wrapJsonRuntimeException(
+              () ->
+                  ProtocolFactory.jsonifiableAdaptableFromJson(JsonFactory.newObject(decodedData)));
+      return decodedJsonifiableAdaptable;
+
+    } else {
+      String data = payloadJson.getValue("data").get().toString();
+      return DittoJsonException.wrapJsonRuntimeException(
+          () -> ProtocolFactory.jsonifiableAdaptableFromJson(JsonFactory.newObject(data)));
+    }
+  }
+
+  @Override
+  public List<Adaptable> map(final ExternalMessage message) {
+    // extract message as String
+    final String payload = extractPayloadAsString(message);
+    if (validatePayload(payload)) {
+      try {
+        JsonifiableAdaptable adaptable = extractData(payload);
+        return singletonList(ProtocolFactory.newAdaptableBuilder(adaptable).build());
+      } catch (Throwable e) {
+         e.printStackTrace();
+      }
+    }
+    return null;
+  }
+
+  @Override
+  public DittoHeaders getAdditionalInboundHeaders(ExternalMessage message) {
+    return DittoHeaders.empty();
+  }
+
+  @Override
+  public List<ExternalMessage> map(final Adaptable adaptable) {
+    return List.of(
+        ExternalMessageFactory.newExternalMessageBuilder(getExternalDittoHeaders(adaptable))
+            .withTopicPath(adaptable.getTopicPath())
+            .withText(getJsonString(adaptable))
+            .asResponse(isResponse(adaptable))
+            .asError(isError(adaptable))
+            .build());
```

Comment:
So the outbound CloudEvents mapping is not part of this PR?
I don't think that it is a good idea to split that up.

Because if we merge this PR, this functionality will be in the code - while only half of the "CloudEvents" mapping functionality is present.

Please also provide the outbound CloudEvent mapping as part of this PR.